diff --git a/lvgl/src/core/lv_refr.c b/lvgl/src/core/lv_refr.c
index 1a56fed0b..2112cf7ce 100644
--- a/lvgl/src/core/lv_refr.c
+++ b/lvgl/src/core/lv_refr.c
@@ -18,6 +18,7 @@
 #include "../draw/lv_draw.h"
 #include "../font/lv_font_fmt_txt.h"
 #include "../extra/others/snapshot/lv_snapshot.h"
+#include "stm32g0xx_hal.h"
 
 #if LV_USE_PERF_MONITOR || LV_USE_MEM_MONITOR
     #include "../widgets/lv_label.h"
@@ -500,6 +501,7 @@ static void lv_refr_join_area(void)
  */
 static void refr_invalid_areas(void)
 {
+    HAL_GPIO_WritePin(GPIOA, GPIO_PIN_10, GPIO_PIN_SET);
     px_num = 0;
 
     if(disp_refr->inv_p == 0) return;
@@ -536,6 +538,7 @@ static void refr_invalid_areas(void)
     }
 
     disp_refr->rendering_in_progress = false;
+    HAL_GPIO_WritePin(GPIOA, GPIO_PIN_10, GPIO_PIN_RESET);
 }
 
 /**
